### ISSUE TITLE
Linter disallows block indentation indicators

### ISSUE THREAD

**Issue by carterbox** [adrienverge/yamllint#793]
According to the YAML spec, you can add an integer at the start of a block multi-line string to indicate the indent level of the block. e.g.

```yaml

example: |6
        This is an indented first line.

      The rest of rest of the string is meant
      to start after 6 spaces. More text. etc.
```

https://yaml.org/spec/1.2.2/#8111-block-indentation-indicator

However, yamllint raises the following syntax error when it detects an integer in the block header.

```
  5:5       error    syntax error: expected <block end>, but found '<scalar>' (syntax)
```

**Comment by adrienverge** [comment]
Hello, which version are you using?

I can't reproduce the syntax error:
```bash
$ yamllint --version
yamllint 1.37.1

$ yamllint - <<EOM
example: |6
        This is an indented first line.

      The rest of rest of the string is meant
      to start after 6 spaces. More text. etc.
EOM
stdin
  1:1       warning  missing document start "---"  (document-start)
```
